include/time.h

```diff
@@ -200,6 +200,8 @@ FAR struct tm *localtime_r(FAR const time_t *timep, FAR struct tm *result);
 
 size_t strftime(FAR char *s, size_t max, FAR const char *format,
                 FAR const struct tm *tm) strftimelike(3);
+FAR char *strptime(FAR const char *s, FAR const char *format,
+                   FAR struct tm *tm);
```

Comment:
```suggestion
#ifdef _XOPEN_SOURCE
FAR char *strptime(FAR const char *s, FAR const char *format,
                   FAR struct tm *tm);
#endif
```

Reply:
The convention of _XOPEN_SOURCE, GNU_SOURCE... is never used in nuttx before. It's better to convert the usage in one patch If this convention is our target.

Reply:
agree